src/overlays/actors/ovl_En_Recepgirl/z_en_recepgirl.c

```diff
@@ -28,38 +32,192 @@ const ActorInit En_Recepgirl_InitVars = {
+        for (i = 0; i < ARRAY_COUNT(sEyeTextures); i++) {
+            sEyeTextures[i] = Lib_SegmentedToVirtual(sEyeTextures[i]);
+        }
+        texturesDesegmented = true;
+    }
+
+    this->eyeTexIndex = 2;
+
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        this->actor.textId = 0x2ADC; // hear directions again?
+    } else {
+        this->actor.textId = 0x2AD9; // "Welcome..."
+    }
+
+    EnRecepgirl_SetupWait(this);
+}
+
+void EnRecepgirl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnRecepgirl_UpdateEyes(EnRecepgirl* this) {
+    if (this->eyeTexIndex != 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex == 4) {
+            this->eyeTexIndex = 0;
+        }
+    } else if (Rand_ZeroOne() < 0.02f) {
+        this->eyeTexIndex++;
+    }
+}
+
+void EnRecepgirl_SetupWait(EnRecepgirl* this) {
+    if (this->skelAnime.animation == &D_06001384) {
+        Animation_MorphToPlayOnce(&this->skelAnime, &D_0600AD98, 5.0f);
+    }
+    this->actionFunc = EnRecepgirl_Wait;
+}
+
+void EnRecepgirl_Wait(EnRecepgirl* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime) != 0) {
+        if (this->skelAnime.animation == &D_0600A280) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &D_0600AD98, 5.0f);
+        } else {
+            Animation_MorphToLoop(&this->skelAnime, &D_06009890, -4.0f);
+        }
+    }
+
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        EnRecepgirl_SetupTalk(this);
+    } else if (Actor_IsActorFacingLink(&this->actor, 0x2000)) {
+        func_800B8614(&this->actor, globalCtx, 60.0f);
+        if (Player_GetMask(globalCtx) == PLAYER_MASK_KAFEIS_MASK) {
+            this->actor.textId = 0x2367; // "... doesn't Kafei want to break off his engagement ... ?"
+        } else if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+            this->actor.textId = 0x2ADC; // hear directions again?
+        } else {
+            this->actor.textId = 0x2AD9; // "Welcome..."
+        }
+    }
+}
+
+void EnRecepgirl_SetupTalk(EnRecepgirl* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &D_0600A280, -4.0f);
+    this->actionFunc = EnRecepgirl_Talk;
+}
+
+void EnRecepgirl_Talk(EnRecepgirl* this, GlobalContext* globalCtx) {
+    u8 temp_v0_2;
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->skelAnime.animation == &D_0600A280) {
+            Animation_PlayLoop(&this->skelAnime, &D_06001384);
+        } else if (this->skelAnime.animation == &D_0600AD98) {
+            if (this->actor.textId == 0x2ADA) { // "The room on your left is The Mayor's office [...]" (meeting ongoing)
+                Animation_MorphToPlayOnce(&this->skelAnime, &D_06000968, 10.0f);
+            } else {
+                Animation_MorphToLoop(&this->skelAnime, &D_06009890, 10.0f);
+            }
+        } else if (this->actor.textId ==
+                   0x2ADA) { // "The room on your left is The Mayor's office [...]" (meeting ongoing)
```

Comment:
Got some strange formatting because of the long text comment. Could you perhaps shorten it, or move it a line up?

Reply:
Hmm, neither option looks very good. I agree the formatting isn't much good either, though.

Reply:
Oh, I know what to do, and it'll make it more symmetrical as well: split the else and if.

Reply:
Drat, there's another one further down this won't work for.